### DISCUSSION TITLE
Mouse Interactivity

### DISCUSSION THREAD

**Discussion by miraclx** [tarkah/tickrs#80] [Ideas]
As mentioned here https://github.com/tarkah/tickrs/issues/72#issuecomment-780708511 and here https://github.com/tarkah/tickrs/discussions/30#discussioncomment-387124, mouse interactivity is a cool feature to add, though currently, it's optionally disabled here #74.

Let's lay out ideas on how this would work when implemented.

I, for one, think the tooltip mentioned here https://github.com/tarkah/tickrs/discussions/30#discussioncomment-387124 is nice.

We could maybe draw greyed-out x-y intersecting lines on that point to show the coordinates of the selection.

I'm quite torn on whether this interaction should be `on:hover` or `on:select`.

Another special feature might be selecting custom time frames, so a <kbd>ctrl</kbd>+<kbd>mouse drag</kbd> would select a custom range on the x-axis within the current time frame and zoom into that as a custom time frame.

Which would then add a new entry in the time frame tabs view in a format like `02-16 18:15 - 02-18 19:15`. I don't think we'd need to have more than one custom time frame so it's easier to work with.

Also, the special <kbd>M</kbd> *(in caps maybe, because it's a privileged key-binding?)*, can, as mentioned here https://github.com/tarkah/tickrs/issues/72#issuecomment-780705510, disable mouse interactivity while the app stays open.

**Comment by tarkah** [comment]
Thanks for putting your thoughts down here! I think I'm going to try and tackle this after I roll out the Kagi chart type I'm working on (this will be SUPER useful for that since time isn't linear across the x-axis for that). 

I think on hover would be best, with the option of toggling mouse interactivity on / off.

"Zoom" would be a cool phase 2 of this. I have scrolling reliably implemented now (from the Kagi chart) and I think thatd the hard part of doing a zoom.

The big caveat behind all of this is that I don't have super high resolution behind the x,y coordinates of the mouse location. It only gives me coordinates in Rows and Columns. But since we use Braille characters for each cell, we actually have (Columns x 2) x  (Rows x 4) resolution when we graph things. So there will always be some inaccuracies / approximations if that makes sense. This wont be an issue for Candles and Kagi since each "candle / trend" takes up at least one cell of width, so we can always show the relevant info for that intersection accurately.
